File: hang.liu/1304.js
var sumZero = function(n) {
    var result = n % 2 === 0 ? [] : [0]
    
    n /= 2
    
    while(n >= 1) {
        result.push(n, 0 - n)
        n -= 1
    }
    
    return result
};

File: xiang.li/1304.java
class Solution {
    public int[] sumZero(int n) {
        int[] a = new int[n];
        int i;
        for(i = 0;i < n / 2;i++) {
            a[i] = i + 1;
        }
        if (n % 2 != 0) {
            a[i] = 0;
            i++;
        }
        for(int j = 0;i + j < n;j++) {
            a[i + j] = 0 - j - 1;
        }
        return a;
    }
}
